controllers/reconcile.go

```diff
@@ -475,6 +577,103 @@ func populateVolumes(d *apps.Deployment, frontend *crd.Frontend, frontendEnviron
 	d.Spec.Template.Spec.Volumes = volumes
 }
 
+// ObjectStoreBucket represents the configuration for an object storage bucket.
+type ObjectStoreBucket struct {
+	AccessKey *string
+	SecretKey *string
+	Name      string
+	Region    *string
+	Endpoint  *string
+	Port      *string
+	TLS       *bool
+}
+
+// ExtractBucketConfigFromSecret extracts ObjectStoreBucket configuration for a specific bucket name
+// from a given Kubernetes Secret. It searches for the bucket name either directly
+// in the secret's Data map under the "bucket" key, or within the
+// "clowder/bucket-names" annotation (if it's a comma-separated list).
+//
+// It returns the ObjectStoreBucket configuration and an error if the secret data is
+// invalid or incomplete for the specified bucket, or if the bucket name isn't found.
+func ExtractBucketConfigFromSecret(secrets []v1.Secret, targetBucketName string) (*ObjectStoreBucket, error) {
+	if len(secrets) == 0 {
+		return nil, fmt.Errorf("no secrets provided to search for bucket '%s'", targetBucketName)
+	}
+
+	clowderBucketNamesAnnotation := "clowder/bucket-names"
+
+	for _, secret := range secrets {
+		currentSecret := secret
+
+		// Check if essential credential keys are present in the current secret's Data
+		requiredCredentialKeys := []string{
+			"aws_access_key_id",
+			"aws_secret_access_key",
+		}
+
+		credentialsPresent := true
+		for _, key := range requiredCredentialKeys {
+			if _, ok := currentSecret.Data[key]; !ok {
+				credentialsPresent = false
+				break
+			}
+		}
+		if !credentialsPresent {
+			continue
+		}
+
+		bucketNameFoundInThisSecret := false
+
+		// Check for "bucket" key in secret.Data
+		if secretBucketData, ok := currentSecret.Data["bucket"]; ok {
+			if string(secretBucketData) == targetBucketName {
+				bucketNameFoundInThisSecret = true
+			}
+		}
+
+		// If not found in Data, check "clowder/bucket-names" annotation
+		if !bucketNameFoundInThisSecret {
+			if annoValue, ok := currentSecret.Annotations[clowderBucketNamesAnnotation]; ok {
+				annotatedBucketNames := strings.Split(annoValue, ",")
+				for _, name := range annotatedBucketNames {
```

Comment:
You can sue the slices util functions to lookup things in arrays: https://pkg.go.dev/slices#Contains, https://pkg.go.dev/slices#ContainsFunc